doofinder-for-woocommerce/includes/class-setup-wizard.php

```diff
@@ -944,27 +968,27 @@ private function process_step_1($processing = false)
 		if (!isset($api_key) || !$api_key || !isset($api_host) || !$api_host || !isset($admin_endpoint) || !$admin_endpoint || !isset($search_endpoint) || !$search_endpoint) {
 
 			if (!isset($api_key) || !$api_key) {
-				$this->errors['wizard-step-1']['api-key'] = __('API key is missing.', 'woocommerce-doofinder');
-				setcookie("doofinderError[wizard-step-1][api-key]", __('API key is missing.', 'woocommerce-doofinder'), 0, '/');
-				$this->log->log('Processing Wizard Step 1 - Error -  API key is missing.');
+				$this->errors['wizard-step-2']['api-key'] = __('API key is missing.', 'woocommerce-doofinder');
+				setcookie("doofinderError[wizard-step-2][api-key]", __('API key is missing.', 'woocommerce-doofinder'), 0, '/');
+				$this->log->log('Processing Wizard Step 2 - Error -  API key is missing.');
 			}
 
 			if (!isset($api_host) || !$api_host) {
-				$this->errors['wizard-step-1']['api-host'] = __('API host is missing.', 'woocommerce-doofinder');
-				setcookie("doofinderError[wizard-step-1][api-host]", __('API host is missing.', 'woocommerce-doofinder'), 0, '/');
-				$this->log->log('Processing Wizard Step 1 - Error - API host is missing.');
+				$this->errors['wizard-step-2']['api-host'] = __('API host is missing.', 'woocommerce-doofinder');
+				setcookie("doofinderError[wizard-step-2][api-host]", __('API host is missing.', 'woocommerce-doofinder'), 0, '/');
+				$this->log->log('Processing Wizard Step 2 - Error - API host is missing.');
 			}
 
 			if (!isset($admin_endpoint) || !$admin_endpoint) {
-				$this->errors['wizard-step-1']['admin-endpoint'] = __('Admin endpoint is missing.', 'woocommerce-doofinder');
-				setcookie("doofinderError[wizard-step-1][admin-endpoint]", __('Admin endpoint is missing.', 'woocommerce-doofinder'), 0, '/');
-				$this->log->log('Processing Wizard Step 1 - Error - Admin endpoint is missing.');
+				$this->errors['wizard-step-2']['admin-endpoint'] = __('Admin endpoint is missing.', 'woocommerce-doofinder');
+				setcookie("doofinderError[wizard-step-2][admin-endpoint]", __('Admin endpoint is missing.', 'woocommerce-doofinder'), 0, '/');
+				$this->log->log('Processing Wizard Step 2 - Error - Admin endpoint is missing.');
 			}
 
 			if (!isset($search_endpoint) || !$search_endpoint) {
-				$this->errors['wizard-step-1']['search-endpoint'] = __('Search endpoint is missing.', 'woocommerce-doofinder');
-				setcookie("doofinderError[wizard-step-1][search-endpoint]", __('Search endpoint is missing.', 'woocommerce-doofinder'), 0, '/');
-				$this->log->log('Processing Wizard Step 1 - Error - Search endpoint is missing.');
+				$this->errors['wizard-step-2']['search-endpoint'] = __('Search endpoint is missing.', 'woocommerce-doofinder');
+				setcookie("doofinderError[wizard-step-2][search-endpoint]", __('Search endpoint is missing.', 'woocommerce-doofinder'), 0, '/');
+				$this->log->log('Processing Wizard Step 2 - Error - Search endpoint is missing.');

```

Comment:
It's a good time to refactor this checks

Reply:
It was a bit more complex to refactor than I thought, but I think the code is cleaner now.